sorting/tim_sort/tim_sort.cpp

```diff
@@ -18,108 +18,153 @@
     Space O(n)
 
 */
-#include <bits/stdc++.h> 
+#include <iostream>
+#include <algorithm>
+#include <vector>
 using namespace std;
 
-const int RUN_SIZE = 32; // for larger data sets we can use 64 as the run size
+const int RUN_SIZE = 32; // For larger data sets we can use 64 as the run size.
+/*
+    Insertion Sort
+
+    Algorithm for perfInsSort:
+
+    Step 1 - Iterate through the array from the index (start + 1) to stop.
+    Step 2 - Select the current element as the key for insertion.
+    Step 3 - Compare the key with elements in the sorted part of the array (start to the previous index).
+    Step 4 - Shift larger elements one position to the right to make space for the key.
+    Step 5 - Insert the key into its correct position in the sorted portion of the array.
+    Step 6 - Repeat this process for every element from (start + 1) to stop.
 
-// Insertion Sort
-void perfInsSort(vector<int>&arr, int start, int stop)
-{
-    for (int idx = start + 1; idx <= stop; ++idx)
-    {
+    Time Complexity Analysis:
+
+    Best    Case O(n)
+    Average Case O(n^2)
+    Worst   Case O(n^2)
+
+
+    Space O(1)- As the sorting is done in-place.
+*/
+void perfInsSort(vector<int>&arr, int start, int stop) {
+    for (int idx = start + 1; idx <= stop; ++idx) {
         int current = arr[idx];
         int prev = idx - 1;
-        while (prev >= start && arr[prev] > current)
-        {
+        while (prev >= start && arr[prev] > current) {
             arr[prev + 1] = arr[prev];
             prev--;
         }
         arr[prev + 1] = current;
     }
 }
 
-// Merge Sort Helper Function
-void mergeSegments(vector<int>& arr, int left, int middle, int right)
-{
-    int len1 = middle - left + 1;
-    int len2 = right - middle;
+/*
+    Merge Sort Helper Function
+
+    Algorithm for mergeSegments:
+
+    Step 1 - Divide the array into two segments (left and right) by finding the middle index.
+    Step 2 - Create two temporary arrays to hold the values of the left and right segments.
+    Step 3 - Copy the values from the original array to the temporary arrays.
+    Step 4 - Merge the two temporary arrays back into the original array by comparing their elements.
+    Step 5 - Insert the smaller of the two elements at each step and move the index.
+    Step 6 - Once one array is exhausted, insert the remaining elements of the other array.
+
+    Time Complexity Analysis:
+
+    Best    Case O(n)
+    Average Case O(n*log(n))
+    Worst   Case O(n*log(n))
+
+    Space O(n)- As temporary arrays are created for merging.
+*/
+void mergeSegments(vector<int>& arr, int leftSize, int middle, int rightSize) {
```

Comment:
I think you refactored the wrong variables!?
Please check

Reply:
I though you ment the function variables 
I will change the len1 len2 to leftSize and rightSize 
my bad sorry

Reply:
@adofm More than what I say, try making sense of why I'm suggesting it! Might help

Reply:
done